chalicelib/checks/higlass_checks.py

```diff
@@ -0,0 +1,1241 @@
+            if time.time() - start_time > 270:
+                time_expired = True
+                break
+
+            # If a particular accession was chosen, skip the others
+            if kwargs['file_accession'] and file_accession != kwargs['file_accession']:
+                continue
+
+            static_content_section = target_files_by_ga[ga][ file_accession ]
+
+            # Post a new Higlass viewconf using the file list
+            higlass_title = "{acc} - Higlass Viewconfig".format(acc=file_accession)
+
+            post_viewconf_results = post_viewconf_to_visualization_endpoint(connection, ref_files, [file_accession], ff_auth, headers, higlass_title, file_info['description'])
+
+            if post_viewconf_results["error"]:
+                action_logs['failed_to_create_viewconf'][file_accession] = post_viewconf_results["error"]
+                continue
+
+            # Create a new static content section with the description = "auto_generated_higlass_view_config" and the new viewconf as the content
+            # Patch the ExpSet static content
+            successful_patch, patch_error =  add_viewconf_static_content_to_file(connection, file_accession, post_viewconf_results["view_config_uuid"],
+            static_content_section,
+            "tab:higlass")
+
+            if not successful_patch:
+                action_logs['failed_to_patch_file'][file_accession] = patch_error
+                continue
+
+            action_logs['new_view_confs_by_file'][file_accession] = post_viewconf_results["view_config_uuid"]
+
+    action.status = 'DONE'
+    action.output = action_logs
+
+    target_files_by_ga = gen_check_result['full_output'].get('target_files', {})
+    file_count = sum([len(target_files_by_ga[ga]) for ga in target_files_by_ga])
+    action.progress = "{completed} out of {possible} files".format(
+        completed=len(action_logs["new_view_confs_by_file"].keys()),
+        possible=file_count
+    )
+    return action
+
+@check_function(expset_accession=None)
+def check_expsets_processedfiles_for_higlass_viewconf(connection, **kwargs):
+    """ Check to generate Higlass view configs on Fourfront for Experiment Sets Processed Files (and Processed Files in Experiment Sets.)
+
+        Args:
+            connection: The connection to Fourfront.
+            **kwargs, which may include:
+                expset_accession: Only check this expset.
+
+        Returns:
+            check results object.
+    """
+
+    check = init_check_res(connection, 'check_expsets_processedfiles_for_higlass_viewconf')
+    check.full_output = {}
+    check.action = 'patch_expsets_processedfiles_for_higlass_viewconf'
+
+    if kwargs['expset_accession']:
+        expsets_by_accession = {
+            kwargs['expset_accession'] : ff_utils.get_metadata(kwargs['expset_accession'], key=connection.ff_keys, ff_env=connection.ff_env, add_on="frame=embedded")
+        }
+    else:
+        fields_to_include = "&field=" + "&field=".join(
```

Comment:
The first `&field=` is unnecessary. Doesn't seem to break anything but maybe remove